doc/scaling_spire.md

```diff
@@ -1,43 +1,92 @@
+A SPIRE deployment has the capacity to be changed in size or scale to accommodate a growing amount of workloads. A SPIRE deployment is composed of a number of one or more SPIRE Servers that share a replicated datastore, or conversely, a set of SPIRE servers in the same trust domain, and at least one SPIRE Agent, but typically more than one.
 
-It is important to note when scaling the number servers horizontally, be it for high availability or load distribution purposes, that in any configuration of two or more SPIRE Servers, all servers share the same datastore. Each server maintains its own CA however, which may be self-signed or an intermediate off of a shared root authority (i.e. when configured with an UpstreamAuthority).
+Deployments range in size. A single SPIRE Server may accommodate a number of Agents and Workload Registration entries. A scale sizing consideration is that with one SPIRE Server per SPIRE deployment, the memory and CPU consumption tends to grow proportionally to the number of Workloads in the deployment. A single SPIRE Server also represents a single point of failure.
 
-# Topologies
+To support larger numbers of Agents and Workloads within a given deployment (tens of thousands or hundreds of thousands of nodes), the number of SPIRE Servers can be scaled horizontally. With multiple servers, the load is distributed. In addition to additional capacity, the use of more than one SPIRE Servers eliminates single points of failure to achieve high availability.
 
-There are three main SPIRE deployment topologies; single trust domain, nested SPIRE, and federated SPIRE. Multiple factors should be considered when choosing one topology design over another.
+## SPIRE Servers in High Availability Mode
 
-A common motivation for a single overarching trust domain is to issue identities from a single Certificate Authority, as it reduces the number of components to manage. However, when deploying a single SPIRE trust domain to span regions, platforms, and cloud provider environments, there is a level of complexity associated with managing a shared datastore across geographically dispersed locations or across cloud provider boundaries. Under these circumstances a solution to address the use of a shared datastore is to configure SPIRE Servers in a nested topology.
+Applicable to all topologies, when scaling the number of servers horizontally, be it for high availability or load distribution purposes, configure all servers in the deployment to share the same datastore (with the exception of Nested topology for sets of SPIRE Servers in different levels).
```

Comment:
A simpler way to say this would be:

_To scale the SPIRE Server horizontally, be it for high availability or load distribution purposes, configure all servers in same trust domain to read and write to the same shared datastore._